components/datetime/src/time_zone.rs

```diff
@@ -463,11 +463,45 @@ impl TimeZoneFormatter {
         })
     }
 
-    icu_provider::gen_any_buffer_constructors!(
-        locale: include,
-        options: TimeZoneFormatterOptions,
-        error: DateTimeError
-    );
+    /// Load generic non location long format for timezone. For example, Pacific Time.
+    #[cfg(feature = "data")]
+    pub fn with_generic_non_location_long(
```

Comment:
Nit: I think I prefer `insert_` as the prefix because `with_` usually refers to something immutable in Java/C++ (sorry, I know I suggested it)

```suggestion
    pub fn insert_generic_non_location_long(
```

Reply:
`insert` sounds a lot like a collection

Reply:
yeah, I prefer `with` (or potentially `include`)